ee/desktop/runner/runner.go

```diff
@@ -800,3 +819,42 @@ func removeFilesWithPrefix(folderPath, prefix string) error {
 		return os.Remove(path)
 	})
 }
+
+// IsAppindicatorEnabled iterates over known appindicator gnome-extensions for each user
+// and returns true the first time it finds one enabled, false if none found enabled
+func IsAppindicatorEnabled(ctx context.Context) bool {
+	uids, err := consoleuser.CurrentUids(ctx)
+	if err != nil {
+		return false
+	}
+
+	var extensions = []string{
+		"ubuntu-appindicators@ubuntu.com",
+		// appindicator that ships with NixOS
+		"appindicatorsupport@rgcjonas.gmail.com",
+	}
+
+	for _, extension := range extensions {
+		for _, uid := range uids {
+			ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+			defer cancel()
+
+			cmd := exec.CommandContext(ctx, "/usr/bin/gnome-extensions", "show", extension)
+
+			// gnome seems to do things through this env
+			cmd.Env = append(cmd.Env, fmt.Sprintf("XDG_RUNTIME_DIR=%s", fmt.Sprintf("/run/user/%s", uid)))
+
+			if err := runas.SetCmdToExecAsUser(ctx, uid, cmd); err != nil {
```

Comment:
I don't think you need this with `XDG_RUNTIME_DIR` set

Reply:
after some testing, this is required when running as root, in face the gnome-extension check fails for me on my ubuntu machine when run as root

Reply:
when running as user, we don't need to set the XDG_RUNTIME_DIR

Reply:
This confuses me. At least my testing found the opposite -- that if I ran as root, with `XDG_RUNTIME_DIR` it was enough. I think it's worth figuring out which of us is correct, since if you're right, we'll need to update the flare i wrote.

Reply:
I get this error on ubuntu. Could you check again or maybe @RebeccaMahany could you give it a shot?

```

/tmp/flare 
❯  sudo /usr/local/kolide-k2/bin/launcher flare
{"caller":"main.go:31","msg":"Launcher starting up","revision":"dba729447ed40e56783695a2cbf0ece230cdb885","severity":"info","ts":"2023-09-12T20:41:33.046304966Z","version":"1.0.16"}
{"caller":"main.go:33","msg":"Launcher starting up","revision":"81def1ba72095207174c5cf91e7ae7290619b312","severity":"info","ts":"2023-09-12T20:41:33.057998277Z","version":"1.0.16-25-g81def1b"}
{"caller":"main.go:188","err":"not rolling out new TUF to channel beta that should still use legacy autoupdater","msg":"could not check out latest launcher, will fall back to old autoupdate library","severity":"error","ts":"2023-09-12T20:41:33.058059875Z"}

/tmp/flare 
❯  unzip -q ./kolide_agent_flare_report_01HA5J1F4229MQG19MB8YNX7J7.zip 

/tmp/flare 
❯  cat ./Gnome\ Extensions/summary.log 
❌	Gnome Extensions: uid:1000: error running gnome-extensions: running "/usr/bin/gnome-extensions show ubuntu-appindicators@ubuntu.com", err is: : exit status 2

```

Reply:
That looks like it's running the wrong version of launcher for what you're trying to test, since it's running the gnome extensions checkup that you deleted?

I pulled the current commit af2106... and I ran `sudo LAUNCHER_SKIP_UPDATES=true ./build/launcher flare` -- my appindicator summary.log reports `No appindicator is enabled, cannot display desktop`, which isn't true.

Reply:
@RebeccaMahany , apologies for not being more specific. I was wondering if the gnome extension check on the current nighlty build worked for you.

Reply:
but also yes, the appindicator check is broken ... fixing it

Reply:
Huh. I'm testing again and I cannot recreate my path. I cannot get this to work without having user permissions, so I think you're right. Empirically I also need to set `XDG_RUNTIME_DIR`, but that's not enough.

Reply:
added `XDG_RUNTIME_DIR` env var back to cmd and working as expected (when run as user)